test/pageobjects/PassportUploadImageGuide.js

```diff
@@ -50,11 +51,23 @@ export default class PassportUploadImageGuide extends BasePage {
     // Input here cannot use the uploadInput() function above
     const input = this.$('.onfido-sdk-ui-CustomFileInput-input')
     const pathToTestFiles = '../resources/'
-    // This will detect local file, ref: https://www.browserstack.com/automate/node#enhancements-uploads-downloads
-    this.driver.setFileDetector(new remote.FileDetector())
-    const sendKeysToElement = input.sendKeys(
-      path.join(__dirname, pathToTestFiles + filename)
-    )
+    //the below line...if safari AND Local, ignore it...otherwise file uploads will fail.
+    if (browserName === 'safari' && isRemoteBrowser === false) {
+      console.log(
+        'Not creating a remote File Detector as I am uploading locally'
+      )
+    } else {
+      // This will detect local file, ref: https://www.browserstack.com/automate/node#enhancements-uploads-downloads
+      this.driver.setFileDetector(new remote.FileDetector())
+    }
+    const filePath = path.join(__dirname, pathToTestFiles + filename)
+    let sendKeysToElement
+    try {
+      console.log(`File being uploaded is ${filePath}`)
+      sendKeysToElement = input.sendKeys(path.join(filePath))
+    } catch (err) {
+      console.log('Just logging that I have caught an exception on upload')
+    }
```

Comment:
This could be a utils function and be used here and in `DocumentUpload.js`

Reply:
Cool, I'll take a look at refactoring that.